lib/openstax/aws/stack_factory.rb

```diff
@@ -80,11 +81,17 @@ def cycle_if_different_parameter(name=nil, &block)
       attributes[:cycle_if_different_parameter] = block.present? ? block.call : name
     end
 
+    def tag(key, value)
+      raise 'The first argument to `tag` must not be blank' if key.blank?
```

Comment:
this raise is done up above too, should only be one of this sort of thing, perhaps in a private method?

Reply:
Here the `Tag` object isn't made yet so seems better to freak out sooner.